micro-rdk/src/common/conn/viam.rs

```diff
@@ -636,6 +639,10 @@ where
                 self.max_concurrent_connections as usize,
             ),
             robot_config: &config,
+            local_signaling_server: Arc::new(SignalingServer::new(
```

Comment:
We need to adjust the default `max_concurrent_connections`, and probably also disable local signaling if the value we are provided is too small, indicating a very resource limited host.